geospaas/base_viewer/views.py

```diff
@@ -56,55 +43,42 @@ def post(self, request, *args, **kwargs):
             self.forms[counter] = element_form_class(request.POST)
 
         self.validation_and_error_check()
-        # modify attributes based on self.form
-        return self.sorting_func(request)
+
+        # modify attributes based on the forms cleaned data
+        self.filtering_func(request)
+        return self.final_rendering(request)
 
     def validation_and_error_check(self):
         for element_forms in self.forms:  # for loop for making clean data by is_valid() method
             element_forms.is_valid()
-            for errorField in element_forms.errors: # temporary error message
+            for errorField in element_forms.errors:  # temporary error message
                 print(f"errorField")
 
-    def sorting_func(self, request):
-        self.ds = CatalogDataset.objects.all()
-        self.filtering_func()
-        return self.final_rendering(request)
-
+    def filtering_func(self, request):
```

Comment:
Why do you need "func" in function name!? Of course it is a "function", what else?
```suggestion
    def filter_datasets(self, request):
```

Reply:
Done